plugins-ui/src/modules/review/components/leave-review/LeaveReview.tsx

```diff
@@ -6,26 +6,51 @@ import { useReviews } from "../../context/ReviewProvider";
 import StarContainer from "@/modules/shared/star-container/StartContainer";
 
 const LeaveReview = () => {
-  const { pluginId, addReview } = useReviews();
+  const { pluginId, addReview, reviewsMap } = useReviews();
 
   const [input, setInput] = useState("");
   const [rating, setRating] = useState(0);
   const [canReview, setCanReview] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(false);
```

Comment:
I suggest using `react-hook-form` instead of managing multiple `useState` calls like input, rating, and isUpdating. It can greatly simplify form handling and improve maintainability.

Reply:
check now